src/wasm-interpreter.h

```diff
@@ -4300,11 +4607,105 @@ class ModuleRunnerBase : public ExpressionRunner<SubType> {
+    auto funcFlow = self()->visit(curr->func);
+    if (funcFlow.breaking()) {
+      return funcFlow;
+    }
+    // Create a new continuation for the target function.
+    Name func = funcFlow.getSingleValue().getFunc();
+    return Literal(std::make_shared<ContData>(func, curr->type.getHeapType()));
+  }
   Flow visitContBind(ContBind* curr) { return Flow(NONCONSTANT_FLOW); }
+  Flow visitSuspend(Suspend* curr) {
+    if (self()->resuming) {
+      // This is a resume, so we have found our way back to where we
+      // suspended.
+      assert(curr == self()->currContinuation->resumeExpr);
+      // We finished resuming, and will continue from here normally.
+      self()->resuming = false;
+      // We should have consumed all the resumeInfo and all the
+      // restoredValues map.
+      assert(self()->currContinuation->resumeInfo.empty());
+      assert(self()->restoredValuesMap.empty());
+      return Flow();
+    }
+
+    // We were not resuming, so this is a new suspend that we must execute.
+    Literals arguments;
+    Flow flow = self()->generateArguments(curr->operands, arguments);
+    if (flow.breaking()) {
+      return flow;
+    }
 
-  Flow visitSuspend(Suspend* curr) { return Flow(NONCONSTANT_FLOW); }
-  Flow visitResume(Resume* curr) { return Flow(NONCONSTANT_FLOW); }
+    if (!self()->currContinuation) {
+      trap("no continuation to suspend");
+    }
+    // Copy the continuation (the old one cannot be resumed again).
+    auto old = self()->currContinuation;
+    assert(old->executed);
+    auto new_ = std::make_shared<ContData>(old->func, old->type);
+    // Switch to the new continuation, so that as we unwind, we will save the
+    // information we need to resume it later in the proper place.
+    self()->currContinuation = new_;
+    // We will resume from this precise spot, when the new continuation is
+    // resumed.
+    new_->resumeExpr = curr;
+    return Flow(SUSPEND_FLOW, curr->tag, std::move(arguments));
+  }
+  Flow visitResume(Resume* curr) {
+    Literals arguments;
+    Flow flow = self()->generateArguments(curr->operands, arguments);
+    if (flow.breaking()) {
+      return flow;
+    }
+    flow = self()->visit(curr->cont);
+    if (flow.breaking()) {
+      return flow;
+    }
+
+    // Get and execute the continuation.
+    auto contData = flow.getSingleValue().getContData();
+    if (contData->executed) {
+      trap("continuation already executed");
+    }
+    contData->executed = true;
+    Name func = contData->func;
+    self()->currContinuation = contData;
+    if (contData->resumeExpr) {
+      // There is an expression to resume execution at, so this is not the first
+      // time we run this function. Mark us as resuming, until we reach that
+      // expression.
+      self()->resuming = true;
+    }
+#if WASM_INTERPRETER_DEBUG
+    std::cout << self()->indent() << "resuming func " << func << '\n';
+#endif
+    Flow ret = callFunction(func, arguments);
```

Comment:
The arguments need to go to the `resume`, not to the function (unless this is a brand new continuation, in which case they do go to the function).